src/BlazorWebView/src/Maui/Windows/WinUIWebViewManager.cs

```diff
@@ -11,11 +16,68 @@ namespace Microsoft.AspNetCore.Components.WebView.Maui
 	/// </summary>
 	public class WinUIWebViewManager : WebView2WebViewManager
 	{
-		readonly WebView2Control _nativeWebView2;
+		private readonly WebView2Control _nativeWebView2;
+		private readonly string _hostPageRelativePath;
+		private readonly string _contentRootDir;
 
-		public WinUIWebViewManager(WebView2Control nativeWebView2, IWebView2Wrapper webview, IServiceProvider services, Dispatcher dispatcher, IFileProvider fileProvider, string hostPageRelativePath) : base(webview, services, dispatcher, fileProvider, hostPageRelativePath)
+		public WinUIWebViewManager(WebView2Control nativeWebView2, IWebView2Wrapper webview, IServiceProvider services, Dispatcher dispatcher, IFileProvider fileProvider, string hostPageRelativePath, string contentRootDir) : base(webview, services, dispatcher, fileProvider, hostPageRelativePath)
 		{
 			_nativeWebView2 = nativeWebView2;
+			_hostPageRelativePath = hostPageRelativePath;
+			_contentRootDir = contentRootDir;
+		}
+
+		protected override void HandleWebResourceRequest(ICoreWebView2WebResourceRequestedEventArgsWrapper eventArgs)
+		{
+			// Unlike server-side code, we get told exactly why the browser is making the request,
+			// so we can be smarter about fallback. We can ensure that 'fetch' requests never result
+			// in fallback, for example.
+			var allowFallbackOnHostPage =
+				eventArgs.ResourceContext == CoreWebView2WebResourceContextWrapper.Document ||
+				eventArgs.ResourceContext == CoreWebView2WebResourceContextWrapper.Other; // e.g., dev tools requesting page source
+
+			// First, call into WebViewManager to see if it has a framework file for this request. It will
+			// fall back to an IFileProvider, but on WinUI it's always a NullFileProvider, so that will never
+			// return a file.
+			if (TryGetResponseContent(eventArgs.Request.Uri, allowFallbackOnHostPage, out var statusCode, out var statusMessage, out var content, out var headers)
+				&& statusCode != 404)
+			{
+				// NOTE: This is stream copying is to work around a hanging bug in WinRT with managed streams
+				var memStream = new MemoryStream();
+				content.CopyTo(memStream);
+				var ms = new InMemoryRandomAccessStream();
+				ms.WriteAsync(memStream.GetWindowsRuntimeBuffer()).AsTask().Wait();
```

Comment:
I see this in a few places, so it might be worth putting in a helper method so that we can catch them all when we get a fix. Maybe an extension method `stream.ToResponseStream()` or something.

Reply:
This is the only place we have this particular code. I'm adding a link to the bug that's tracking it. I don't want to put it in an extension method because that might look "too official." I want this code to be as non-reusable as possible 😄